app/src/main/java/com/example/androidapp/sensors/SensorController.java

```diff
@@ -0,0 +1,161 @@
+package com.example.androidapp.sensors;
+
+import android.annotation.SuppressLint;
+
+import android.content.Context;
+
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+import android.os.SystemClock;
+import android.util.Log;
+
+import com.example.androidapp.DatabaseController;
+import com.example.androidapp.ui.report.ReportFragment;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.List;
+import java.util.TimeZone;
+
+public class SensorController {
+
+    private final Date cDate = new Date();
+    static int stepsCompleted = 0;
+
+    public SensorController(Context context) {
+
+        Log.d("STEPSENSOR: ", "sensor controller created");
+        SensorManager mSensorManager = (SensorManager) context.getSystemService(Context.SENSOR_SERVICE);
+        Sensor mSensorACC = mSensorManager.getDefaultSensor(Sensor.TYPE_LINEAR_ACCELERATION);
+        Sensor mSensorStepDetector = mSensorManager.getDefaultSensor(Sensor.TYPE_STEP_DETECTOR);
+
+        SensorEventListener listener = new StepCounterListener(context);
+            if (mSensorACC != null) {
+                Log.d("STEPSENSOR: ", "mSensorACC not null");
+                mSensorManager.registerListener(listener, mSensorACC, SensorManager.SENSOR_DELAY_NORMAL);
+            } else {
+                Log.d("STEPSENSOR: ", "mSensorACC not  found");
+            }
+            if (mSensorStepDetector != null) {
+                Log.d("STEPSENSOR: ", "mSensorStepDetector not null");
+                mSensorManager.registerListener(listener, mSensorStepDetector, SensorManager.SENSOR_DELAY_NORMAL);
+            } else {
+                Log.d("STEPSENSOR: ", "mSensorStepDetector not found");
+            }
+        }
+
+    public int dailySteps(Context context){
+
+        Calendar calendar = new GregorianCalendar(TimeZone.getTimeZone("GMT"));
+        DateFormat formatter = new SimpleDateFormat("yyyy-MM-dd");
+        formatter.setTimeZone(TimeZone.getTimeZone("GMT+2"));
+        String fDate = formatter.format(calendar.getTime());
+       // String fDate = new SimpleDateFormat("yyyy-MM-dd").format(cDate);
+
+        stepsCompleted=DatabaseController.loadStepsForTheDay(context, fDate);
+        return stepsCompleted;
+    }
+
+    //TODO: method that returns all the steps
+    public int totalSteps(){
```

Comment:
I think this will be removed because I have never seen an app where the overall total steps registered in the app are shown (it does not make sense for the user, right?).